iam-keycloak-migration.sh

```diff
@@ -557,8 +557,20 @@ then
   # Get the name of the keycloak client
   cp4iKeycloakClientId="$(oc get integrationkeycloakclient.keycloak.integration.ibm.com -l app.kubernetes.io/name=ibm-integration-platform-navigator -n "$servicesNamespace" -o jsonpath='{.items[0].spec.client.clientId}')"
 
-  keycloakAdminSecretName="cs-keycloak-initial-admin"
-  keycloakAdminPass="$(oc get secret "$keycloakAdminSecretName" -n "$servicesNamespace" -o jsonpath="{.data.password}" | base64 -d)"
+  keycloakIntegrationUserName="internal-keycloak-user"
+  keycloakIntegrationAdminSecretName="internal-keycloak-user-secret"
+  
+  # Default to the CS values, until we can validate that our internal IKU is fully reconciled
+  secretToFetch="cs-keycloak-initial-admin"
+  keycloakAdminUser="admin"
+  
+  # Check the integration internal IKU is reconciled and if so, use t instead of the CS one
+  integrationIKUPhase="$(oc get integrationkeycloakuser "$keycloakIntegrationUserName" -n "$servicesNamespace" -o jsonpath='{.status.phase}' --ignore-not-found=true)"
+  if [[ "$integrationIKUPhase" == "reconciled" ]]; then
+    secretToFetch=$keycloakIntegrationAdminSecretName
+    keycloakAdminUser="$keycloakIntegrationUserName"
+  fi  
+  keycloakAdminPass="$(oc get secret "$secretToFetch" -n "$servicesNamespace" -o jsonpath="{.data.password}" | base64 -d)"
```

Comment:
Something I've just spotted, more of a nitpick I don't think it's necessary to have quotes around the variables here

```suggestion
  keycloakAdminPass="$(oc get secret $secretToFetch -n $servicesNamespace -o jsonpath="{.data.password}" | base64 -d)"
```

Reply:
Yes, you do want these quotes.

Reply:
Cool, I'll keep them in then!